From b5ba1d0483f1b6d0b5139ded53dd53ca5d165226 Mon Sep 17 00:00:00 2001
From: Mark Pauna <mark.pauna@windriver.com>
Date: Thu, 6 Nov 2008 13:49:44 -0600
Subject: [PATCH] oprofile: export address range symbols for stack crawl

Export symbols which can be used to identify the address ranges of the
syscall handler routines.  These symbols can be used during a stack
backtrace to identify the boundary between a user application's stack
and the system stack.

Signed-off-by: Mark Pauna <mark.pauna@windriver.com>
Signed-off-by: Dave Lerner <dave.lerner@windriver.com>
---
 arch/mips/kernel/scall32-o32.S |    1 +
 arch/mips/kernel/scall64-64.S  |    2 ++
 arch/mips/kernel/scall64-n32.S |    2 ++
 arch/mips/kernel/scall64-o32.S |    1 +
 4 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/arch/mips/kernel/scall32-o32.S b/arch/mips/kernel/scall32-o32.S
index 428725c..393874f 100644
--- a/arch/mips/kernel/scall32-o32.S
+++ b/arch/mips/kernel/scall32-o32.S
@@ -186,6 +186,7 @@ illegal_syscall:
 	sw	t0, PT_R7(sp)
 	j	o32_syscall_exit
 	END(handle_sys)
+EXPORT(ret_from_handle_sys)
 
 	LEAF(mips_atomic_set)
 	andi	v0, a1, 3			# must be word aligned
diff --git a/arch/mips/kernel/scall64-64.S b/arch/mips/kernel/scall64-64.S
index a9e1716..2373f96 100644
--- a/arch/mips/kernel/scall64-64.S
+++ b/arch/mips/kernel/scall64-64.S
@@ -23,6 +23,7 @@
 #ifndef CONFIG_BINFMT_ELF32
 /* Neither O32 nor N32, so define handle_sys here */
 #define handle_sys64 handle_sys
+#define ret_from_handle_sys64 ret_from_handle_sys
 #endif
 
 	.align  5
@@ -123,6 +124,7 @@ illegal_syscall:
 	sd	t0, PT_R7(sp)
 	j	n64_syscall_exit
 	END(handle_sys64)
+EXPORT(ret_from_handle_sys64)
 
 	LEAF(mips_atomic_set)
 	andi	v0, a1, 3			# must be word aligned
diff --git a/arch/mips/kernel/scall64-n32.S b/arch/mips/kernel/scall64-n32.S
index da7f1b6..951fd24 100644
--- a/arch/mips/kernel/scall64-n32.S
+++ b/arch/mips/kernel/scall64-n32.S
@@ -26,6 +26,7 @@
 #ifndef CONFIG_MIPS32_O32
 /* No O32, so define handle_sys here */
 #define handle_sysn32 handle_sys
+#define ret_from_handle_sysn32 ret_from_handle_sys
 #endif
 
 	.align  5
@@ -118,6 +119,7 @@ not_n32_scall:
 	j	handle_sys64
 
 	END(handle_sysn32)
+EXPORT(ret_from_handle_sysn32)
 
 EXPORT(sysn32_call_table)
 	PTR	sys_read			/* 6000 */
diff --git a/arch/mips/kernel/scall64-o32.S b/arch/mips/kernel/scall64-o32.S
index d7cd1aa..03a07d1 100644
--- a/arch/mips/kernel/scall64-o32.S
+++ b/arch/mips/kernel/scall64-o32.S
@@ -172,6 +172,7 @@ not_o32_scall:
 	j	handle_sys64
 #endif
 	END(handle_sys)
+EXPORT(ret_from_handle_sys)
 
 LEAF(sys32_syscall)
 	sltu	v0, a0, __NR_O32_Linux + __NR_O32_Linux_syscalls + 1
-- 
1.6.0.90.g436ed

